drivers: ieee802154: remove dummy implementation of ieee802154_radio_handle_ack() from ieee802154_radio.h

MeisterBob opened the pull request:
When using the IEEE802154 radio interface in `CONFIG_IEEE802154_RAW_MODE`, there is no way to handle ack frames because of the dummy implementation of `ieee802154_radio_handle_ack()` in the ieee802154_radio header file.

Removed the dummy implementation from the header so you can/need to implement it in the application. Added the dummy implementation to the wpan_serial and wpanusb sample as they use `CONFIG_IEEE802154_RAW_MODE`.

rlubos commented on samples/net/wpan_serial/src/main.c:
```diff
@@ -520,6 +520,12 @@ int net_recv_data(struct net_if *iface, struct net_pkt *pkt)
 	return 0;
 }
 
+inline enum net_verdict ieee802154_radio_handle_ack(struct net_if *iface,
```
I don't think you need `inline`, it doesn't match the declaration now.

rlubos commented on samples/net/wpan_serial/src/main.c:
```diff
@@ -520,6 +520,12 @@ int net_recv_data(struct net_if *iface, struct net_pkt *pkt)
 }
 
+inline enum net_verdict ieee802154_radio_handle_ack(struct net_if *iface,
+							   struct net_pkt *pkt)
```
Please fix the alignment as well.

rlubos commented on samples/net/wpanusb/src/wpanusb.c:
```diff
@@ -409,6 +409,12 @@ int net_recv_data(struct net_if *iface, struct net_pkt *pkt)
 	return ret;
 }
 
+inline enum net_verdict ieee802154_radio_handle_ack(struct net_if *iface,
```
Same here.

rlubos reviewed (commented):
The commit message needs to follow specific format, see https://docs.zephyrproject.org/latest/contribute/#commit-guidelines. 

Also, I think that this change should be handled in a single commit - otherwise we break the bisectability of the samples, they won't build with only one of them applied.

Looks good otherwise, just some nits.

MeisterBob commented:
squashed the commits, removed inline and fixed alignment

rlubos commented:
Note, there's a char/line limit for the commit title/body.

MeisterBob commented:
added some linebreaks to the commit message

rlubos commented:
The CI is very picky in terms of commit verification, there's still:
```
9be9a86d1331e7121c195bcd6969b1904f6b7927: author email (Gerhard <joerges@metratec.com>) needs to match one of the signed-off-by entries.
9be9a86d1331e7121c195bcd6969b1904f6b7927: author email (Gerhard <joerges@metratec.com>) does not follow the syntax: First Last <email>.
```
It looks that you need to update your git `user.name` config to your full name (just as used in the signoff), and then update the commit author (`git commit --amend --reset-author --no-edit`)

rlubos reviewed (approved):
LGTM, just some CI issues left.

mt-gjoerges committed:
drivers: ieee802154: remove ieee802154_radio_handle_ack() stub

When using the IEEE802154 radio interface in CONFIG_IEEE802154_RAW_MODE,
there is no way to handle ack frames because of the dummy implementation
of ieee802154_radio_handle_ack() in the ieee802154_radio header file.
Removed the dummy implementation from the header so you can/need to
implement it in the application.

samples: added implementation of ieee802154_radio_handle_ack() to
         wpan_serial and wpanusb sample

Signed-off-by: Gerhard Jörges <joerges@metratec.com>

MeisterBob commented:
that should satisfy the CI
